libvirt.go

```diff
@@ -717,3 +717,27 @@ func NewWithDialer(dialer socket.Dialer) *Libvirt {
 func New(conn net.Conn) *Libvirt {
 	return NewWithDialer(dialers.NewAlreadyConnected(conn))
 }
+
+// NetworkUpdateFixed is a wrapper over NetworkUpdate which swaps `Command` and `Section` when needed.
+// This function must be used instead of NetworkUpdate to be sure that the
+// NetworkUpdate call works both with older and newer libvirtd connections.
+//
+// libvirt on-wire protocol had a bug for a long time where Command and Section
+// were reversed. It's been fixed in newer libvirt versions, and backported to
+// some older versions. This helper detects what argument order libvirtd expects
+// and makes the correct NetworkUpdate call.
+func (l *Libvirt) NetworkUpdateFixed(Net Network, Command NetworkUpdateCommand, Section NetworkUpdateSection, ParentIndex int32, XML string, Flags NetworkUpdateFlags) (err error) {
+	// This is defined in libvirt/src/libvirt_internal.h and thus not available in go-libvirt autogenerated code
+	const virDrvFeatureNetworkUpdateHasCorrectOrder = 16
+	hasCorrectOrder, err := l.ConnectSupportsFeature(virDrvFeatureNetworkUpdateHasCorrectOrder)
+	if err != nil {
+		// TODO: ignore error and force hasCorrectOrder to 0?
```

Comment:
Looking at Libvirt.ConnectSupportsFeature, I'd vote we just propagate the error down like it is now; otherwise, with those classes of errors ((de/en)coding errors, transport layer errors), I'm guessing the subsequent calls to Libvirt.NetworkUpdate will fail for similar reasons.

Reply:
The reason  for the comment is that I was not sure what would happen if the remote end has no implementation for `ConnectSupportsFeature`, or if the connection does not support the feature.
For the former, I now see the method has been there forever (at least libvirt 0.3.0), so this is not going to be a real concern. And thinking more of it, I don't see why an error would be returned instead of `hasCorrectOrder = 0` when the method is present but the feature not supported. So yeah, if there's an error, it's going to be a 'bad' one, and we should return it.